### PR TITLE
feat: opt tab for inspector; localStorage settings

### PR DESCRIPTION
**Pull request by maxkrieger**
*Autostep and show inspector are now togglable across refreshes via localStorage. By default autostep's off and inspector's on.

* We added a simple sortable table with energies of functions. More features to come

image

### CODE REVIEW

File: packages/browser-ui/src/inspector/views/Opt.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  return (
+    <div>
+      <DataTable
+        data={[...constrInfos, ...objInfos]}
+        title={"Optimization"}
+        dense={true}
+        highlightOnHover={true}
+        striped={true}
+        columns={[
+          { name: "Expression", sortable: true, selector: "name" },
```
Comment (hypotext) [inline packages/browser-ui/src/inspector/views/Opt.tsx:33]: change order of `selector` and `sortable` to be consistent?

**Review by hypotext — CHANGES_REQUESTED** [review]
Thanks for this! @wodeni did you want to take a look before we merge?

Some comments:

* Not sure why the text disappears from the diagram when I click on an inspector frame. Would be good to fix this first.
Before:


After: (note selected frame)

https://ci6.googleusercontent.com/proxy/exedg8Ev6mD5XUFcB7pAaF20aiCKw0V9ikLdYpxQqCpi7pzoJRABFLmTOu_Tz3SO6xBzDzgYtrdJMmbX_hNF5Ivw0y1UpT-OIIPOC4vYU_2mv0mrcs9m8rNwVAvAxRpRwMJ7W6g_F7jaR8iJ88ygm-rgn7UnLF8=s0-d-e1-ft#https://user-images.githubusercontent.com/2762356/112903461-89c80700-90b5-11eb-84cb-522eb4d035a9.png

* If `autostep` is on, I am confused by the `opt` values shown for each frame (e.g. the past frames). They do seem to be changing. Are those the actual values for the past step in the optimization? Is the inspector auto-saving the history of values, and if so, how/where is it doing that? If not, what are those values from?

May be confusing to existing users if we have `autostep` off by default. Maybe we should just let them know.

**Comment by wodeni** [comment]
> Thanks for this! @wodeni did you want to take a look before we merge?

Already read through the changes. Looks great. Feel free to merge whenever you are ready 💯 !

**Comment by maxkrieger** [comment]
Text disappearance was always the case not sure if it's accentuated now. I think it's an ID conflict... or maybe global symbols..

**Comment by maxkrieger** [comment]
> * If `autostep` is on, I am confused by the `opt` values shown for each frame (e.g. the past frames). They do seem to be changing. Are those the actual values for the past step in the optimization? Is the inspector auto-saving the history of values, and if so, how/where is it doing that? If not, what are those values from?

The history of the values is in the History object. It's just the states. The past frames are visually different states, no? It may be adding more than one frame due to some sort of eventing quirk... Hmm

Autostep is off by default btw.

**Comment by hypotext** [comment]
> The history of the values is in the History object. It's just the states.

Ok, trying to state my question more clearly:

* I thought the `opt` tab only calculates the `opt` info for the current frame (via `evalFns`, etc) and displays them. It doesn't store the values anywhere.
* However, when I click on past frames, I see different opt values in the tab `opt`.
* Where are those past frames` opt values coming from? 

I guess if the text thing isn't a regression then we don't have to fix it now, though it would be good to have an issue for it.

**Comment by hypotext** [comment]
I also noticed that the `opt` tab doesn't work in conjunction with the `mod` tab, i.e. if I use the `mod` tab to change some values, the opt values are not re-updated. That makes harder to test energies, though we can get to that in a second pass.

**Comment by maxkrieger** [comment]
image

File: packages/browser-ui/src/inspector/views/Opt.tsx (hunk 1 above)
Comment (maxkrieger) [reply packages/browser-ui/src/inspector/views/Opt.tsx:43]: done :D

**Review by hypotext — APPROVED** [review]
LGTM! These features have already been massively useful for me on the `graph-domain` branch. (Another TODO is to, at some point, document the opt tab on the wiki.)

**Pull request merged**